WIP MOSS refactoring

mamhoff opened the pull request:
This is the PR that will actually bring about behaviour changes. Notably, taxable items (line items, shipping rates, and shipments) will have their price adjusted to the correct price including any applicable VAT. There shall be no refunds. 

This is still work in progress. It depends on #685 (which it includes).

mtomov commented on core/spec/models/spree/line_item_spec.rb:
```diff
@@ -4,6 +4,27 @@
+  describe 'initial_price' do
+    it 'is available' do
+      expect(line_item).to respond_to(:initial_price)
+    end
```
unnecessary : ) The tests below would fail anyhow.

The BigDecimal one is also superfluous imo

mtomov commented on core/spec/models/spree/tax/item_adjuster_spec.rb:
```diff
@@ -0,0 +1,29 @@
+    it 'sets adjustable' do
+      expect(adjuster.instance_variable_get(:@item)).to eq(item)
+    end
+  end
```
I'd rather stick to testing only public interface of a class. The below `#adjust` should fail of those are not being correctly set. Otherwise, you introduce coupling between the code and the tests : )

mamhoff commented:
Rebased on https://github.com/solidusio/solidus/pull/685, and removed the work on shipping rates present in RfC Make shipping rates taxable like line items or shipments  (on which this will have to be rebased as well).

mamhoff commented:
After chat on slack with @jhawthorn reverted the instance variable names to names without an underscore (as they're actually used by the classes that use the `TaxHelpers` mix-in.

tvdeyen commented:
That's reasonable and I'm fine with it. Although `@rates_for_items` is not used outside of `ItemAdjuster` and therefore could have the underscore in the name, but I will stop complaining now and won't hold this back anymore. 

Great work. Thanks, @mamhoff

mamhoff commented:
I'm closing this. Much of it is already merged into master, and for all the following work, #989 is the guide.
